cpu/x86/x86_pit.c

```diff
@@ -55,6 +55,8 @@ void x86_pit_set2(unsigned channel, unsigned mode, uint16_t max)
 {
     unsigned old_flags = irq_disable();
     outb(PIT_COMMAND_PORT, ((channel - 1) << 6) | mode | PIT_ACCESS_MODE_LO_HI);
+    /* cppcheck-suppress knownConditionTrueFalse
+     * cppcheck bug - there is no condition here */
```

Comment:
I think this is wrong. The line below should probably be `max & 0xff` instead of `&&`